autogluon_zeroshot/simulation/tabular_predictions.py

```diff
@@ -328,32 +462,36 @@ def load(cls, filename: str):
         dataset_to_models = metadata["dataset_to_models"]
 
         return cls(
-            dataset_to_models=dataset_to_models,
+            tasks_to_models=dataset_to_models,
             output_dir=filename,
         )
 
-    def models_available_in_dataset(self, dataset: str) -> List[str]:
-        # todo handle slice
-        return self.dataset_to_models[self.rename_dict_inv.get(dataset, dataset)]
-
     @property
     def folds(self) -> List[int]:
-        # TODO
-        return list(range(10))
+        first = next(iter(self.tasks_to_models.values()))
+        return sorted(list(first.keys()))
```

Comment:
This change specifically required a lot of changes, as previously folds logic was not working correctly in the lazy variant.